tests/cxx/badinc.cc

```diff
@@ -1237,11 +1235,13 @@ int main() {
   i1_template_method_only_class.s<I2_Class*>(local_i2_class_ptr);
   // IWYU: I1_Class is...*badinc-i1.h
   // IWYU: I1_Class needs a declaration
+  // IWYU: I1_TemplateMethodOnlyClass<I1_Class, I1_Union>::s is...*badinc-i1.h
```

Comment:
All the changes in this file make me nervous... Given how completely impenetrable `badinc.cc` is, I get the feeling you've just patched it after-the-fact to shut the test suite up... Which in turn could indicates that some critical functionality changed behavior. But I can't pretend to know what the basis for those changes are, so maybe you could explain?